Description: Generally we don't need to write to every single block device we
 open, in fact if we do that when scanning we'll make udev think they all
 changed, and it could run "lvm vgscan" again!  When the file descriptor
 is needed for writing, it will be closed and opened again.
Author: Scott James Remnant <scott@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/332270

Index: lvm2-2.02.66/lib/device/dev-md.c
===================================================================
--- lvm2-2.02.66.orig/lib/device/dev-md.c	2009-08-19 11:34:43.000000000 -0400
+++ lvm2-2.02.66/lib/device/dev-md.c	2010-11-12 10:17:19.000000000 -0500
@@ -94,7 +94,7 @@
 	if (size < MD_RESERVED_SECTORS * 2)
 		return 0;
 
-	if (!dev_open(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0)) {
 		stack;
 		return -1;
 	}
Index: lvm2-2.02.66/lib/format1/disk-rep.c
===================================================================
--- lvm2-2.02.66.orig/lib/format1/disk-rep.c	2009-10-27 13:00:46.000000000 -0400
+++ lvm2-2.02.66/lib/format1/disk-rep.c	2010-11-12 10:17:19.000000000 -0500
@@ -415,7 +415,7 @@
 {
 	struct disk_list *dl;
 
-	if (!dev_open(dev))
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	dl = __read_disk(fmt, dev, mem, vg_name);
Index: lvm2-2.02.66/lib/format_pool/disk_rep.c
===================================================================
--- lvm2-2.02.66.orig/lib/format_pool/disk_rep.c	2009-10-27 13:00:46.000000000 -0400
+++ lvm2-2.02.66/lib/format_pool/disk_rep.c	2010-11-12 10:17:19.000000000 -0500
@@ -350,7 +350,7 @@
 {
 	struct pool_list *pl;
 
-	if (!dev_open(dev))
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	if (!(pl = dm_pool_zalloc(mem, sizeof(*pl)))) {
Index: lvm2-2.02.66/lib/format_text/format-text.c
===================================================================
--- lvm2-2.02.66.orig/lib/format_text/format-text.c	2010-04-14 09:09:16.000000000 -0400
+++ lvm2-2.02.66/lib/format_text/format-text.c	2010-11-12 10:17:19.000000000 -0500
@@ -165,7 +165,7 @@
 		  PRIu64, mdac->area.start, mdac->area.size);
 	area = &mdac->area;
 
-	if (!dev_open(area->dev))
+	if (!dev_open_flags(area->dev, O_RDONLY, 1, 0))
 		return_0;
 
 	if (!(mdah = _raw_read_mda_header(fmt, area)))
@@ -437,7 +437,7 @@
 	int noprecommit = 0;
 	struct mda_header *mdah;
 
-	if (!dev_open(dev_area->dev))
+	if (!dev_open_flags(dev_area->dev, O_RDONLY, 1, 0))
 		return_0;
 
 	if (!(mdah = _raw_read_mda_header(fid->fmt, dev_area)))
@@ -464,7 +464,7 @@
 	char *desc;
 	uint32_t wrap = 0;
 
-	if (!dev_open(area->dev))
+	if (!dev_open_flags(area->dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	if (!(mdah = _raw_read_mda_header(fid->fmt, area)))
@@ -1068,7 +1068,7 @@
 	if (mda_free_sectors)
 		*mda_free_sectors = ((dev_area->size - MDA_HEADER_SIZE) / 2) >> SECTOR_SHIFT;
 
-	if (!dev_open(dev_area->dev))
+	if (!dev_open_flags(dev_area->dev, O_RDONLY, 1, 0))
 		return_NULL;
 
 	if (!(mdah = _raw_read_mda_header(fmt, dev_area)))
Index: lvm2-2.02.66/lib/label/label.c
===================================================================
--- lvm2-2.02.66.orig/lib/label/label.c	2009-07-15 16:02:47.000000000 -0400
+++ lvm2-2.02.66/lib/label/label.c	2010-11-12 10:17:19.000000000 -0500
@@ -272,7 +272,7 @@
 		return 1;
 	}
 
-	if (!dev_open(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0)) {
 		stack;
 
 		if ((info = info_from_pvid(dev->pvid, 0)))
@@ -352,7 +352,7 @@
 	struct lvmcache_info *info;
 	int r = 0;
 
-	if (!dev_open(dev)) {
+	if (!dev_open_flags(dev, O_RDONLY, 1, 0)) {
 		if ((info = info_from_pvid(dev->pvid, 0)))
 			lvmcache_update_vgname_and_id(info, info->fmt->orphan_vg_name,
 						      info->fmt->orphan_vg_name,
